main/src/library/MutList.flix

```diff
@@ -801,7 +835,9 @@ namespace MutList {
     ///
     /// Modifying `l` while using an iterator has undefined behavior and is dangerous.
     ///
-    pub def iterator(l: MutList[a, r]): Iterator[a, r] \ Write(r) =
+    /// Andreas: "Function reads from len, so I have added Read(r) effect.
```

Comment:
Question 1: Since the length of an array is fixed (once created) should this operation be pure or impure?
Question 2: What does Flix do?